plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/reporter/Reporter.kt

```diff
@@ -0,0 +1,59 @@
+// Collection of functions that applies output Ktlint reporter
+
+package org.jlleitschuh.gradle.ktlint.reporter
+
+import org.gradle.api.Project
+import org.gradle.api.tasks.JavaExec
+import org.gradle.util.GFileUtils
+import java.io.File
+import net.swiftzer.semver.SemVer
+import org.jlleitschuh.gradle.ktlint.KtlintExtension
+import java.io.FileOutputStream
+import java.io.OutputStream
+
+/**
+ * Apply reporter to the task.
+ */
+internal fun JavaExec.applyReporters(target: Project, extension: KtlintExtension, sourceSetName: String) {
+    if (SemVer.parse(extension.version).compareTo(SemVer(0, 10, 0)) >= 0) {
```

Comment:
Documentation why this version check is necessary would be very helpful.

Reply:
In Koltin you should be able to do something like `SemVer.parse(extension.version) >= SemVer(0, 10, 0)` so you don't need to have the extra `>= 0`.